common/repositories/inventory_snapshots_repository.h

```diff
@@ -68,6 +95,227 @@ class InventorySnapshotsRepository: public BaseInventorySnapshotsRepository {
+			"WHERE b.`time_index` = '{1}' AND b.`character_id` = '{0}' AND a.`character_id` = '{0}')",
+			character_id,
+			timestamp
+		);
+
+		auto results = db.QueryDatabase(query);
+		if (!results.Success()) {
+			return;
+		}
+
+		for (auto row: results) {
+			compare_list.emplace_back(std::pair<int16, uint32>(Strings::ToInt(row[0]), Strings::ToUnsignedInt(row[1])));
+		}
+	}
+
+	static bool SaveCharacterInvSnapshot(Database &db, uint32 character_id)
+	{
+		uint32                          time_index = time(nullptr);
+		std::vector<InventorySnapshots> queue{};
+
+		auto inventory = InventoryRepository::GetWhere(db, fmt::format("`character_id` = '{}'", character_id));
+		if (inventory.empty()) {
+			LogError("Character ID [{}] inventory is empty. Snapshot not created", character_id);
+			return false;
+		}
+
+		for (auto const &i: inventory) {
+			auto snapshot                = NewEntity();
+			snapshot.character_id        = i.character_id;
+			snapshot.item_id             = i.item_id;
+			snapshot.item_unique_id      = i.item_unique_id;
+			snapshot.augment_one         = i.augment_one;
+			snapshot.augment_two         = i.augment_two;
+			snapshot.augment_three       = i.augment_three;
+			snapshot.augment_four        = i.augment_four;
+			snapshot.augment_five        = i.augment_five;
+			snapshot.augment_six         = i.augment_six;
+			snapshot.charges             = i.charges;
+			snapshot.color               = i.color;
+			snapshot.custom_data         = i.custom_data;
+			snapshot.instnodrop          = i.instnodrop;
+			snapshot.ornament_hero_model = i.ornament_hero_model;
+			snapshot.ornament_icon       = i.ornament_icon;
+			snapshot.ornament_idfile     = i.ornament_idfile;
+			snapshot.slot_id             = i.slot_id;
+			snapshot.time_index          = time_index;
+			queue.push_back(snapshot);
+		}
+
+		if (queue.empty()) {
+			LogError("Character ID [{}] inventory is empty. Snapshot not created", character_id);
+			return false;
+		}
+
+		if (!InsertMany(db, queue)) {
+			LogError("Failed to created inventory snapshot for [{}]", character_id);
+			return false;
+		}
+
+		LogInventory("Created inventory snapshot for [{}] with ([{}]) items", character_id, queue.size());
+		return true;
+	}
+
+	static bool RestoreCharacterInvSnapshot(Database &db, uint32 character_id, uint32 timestamp)
+	{
+		InventoryRepository::DeleteWhere(db, fmt::format("`character_id` = '{}'", character_id));
+
+		auto snapshot = GetWhere(db, fmt::format("`character_id` = '{}' AND `time_index` = '{}'", character_id, timestamp));
+		if (snapshot.empty()) {
+			LogError("The snapshot requested could not be found.  Restore failed for character id [{}] @ [{}] failed",
+				character_id,
+				timestamp
+			);
+			return false;
+		}
+
+		std::vector<InventoryRepository::Inventory> queue{};
+		for (auto const &i: snapshot) {
+			auto inventory_entry                = InventoryRepository::NewEntity();
```

Comment:
Just make this `e` vs `inventory_entry` - makes the code needlessly verbose when its a receiver of a complex type